gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -594,68 +594,22 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
 
 // This rule will try to convert the row-to-columnar and columnar-to-row
 // into native implementations.
-case class TransformPostOverrides(isAdaptiveContext: Boolean) extends Rule[SparkPlan] {
+case class TransformPostOverrides() extends Rule[SparkPlan] {
   val columnarConf: GlutenConfig = GlutenConfig.getConf
   @transient private val planChangeLogger = new PlanChangeLogger[SparkPlan]()
 
-  def transformColumnarToRowExec(plan: ColumnarToRowExec): SparkPlan = {
-    if (columnarConf.enableNativeColumnarToRow) {
-      val child = replaceWithTransformerPlan(plan.child)
-
-      if (!PlanUtil.outputNativeColumnarData(child)) {
-        TransformHints.tagNotTransformable(plan, "child is not gluten plan")
-        plan.withNewChildren(plan.children.map(replaceWithTransformerPlan))
-      } else {
-        logDebug(s"ColumnarPostOverrides ColumnarToRowExecBase(${child.nodeName})")
-        val nativeConversion =
-          BackendsApiManager.getSparkPlanExecApiInstance.genColumnarToRowExec(child)
-        val validationResult = nativeConversion.doValidate()
-        if (validationResult.isValid) {
-          nativeConversion
-        } else {
-          TransformHints.tagNotTransformable(plan, validationResult)
-          plan.withNewChildren(plan.children.map(replaceWithTransformerPlan))
-        }
-      }
-    } else {
-      plan.withNewChildren(plan.children.map(replaceWithTransformerPlan))
-    }
-  }
-
-  def replaceWithTransformerPlan(plan: SparkPlan): SparkPlan = plan match {
-    case plan: RowToColumnarExec =>
-      val child = replaceWithTransformerPlan(plan.child)
+  def replaceWithTransformerPlan(plan: SparkPlan): SparkPlan = plan.transformDown {
+    case RowToColumnarExec(child) =>
       logDebug(s"ColumnarPostOverrides RowToColumnarExec(${child.getClass})")
       BackendsApiManager.getSparkPlanExecApiInstance.genRowToColumnarExec(child)
-    // The ColumnarShuffleExchangeExec node may be the top node, so we cannot remove it.
-    // e.g. select /* REPARTITION */ from testData, and the AQE create shuffle stage will check
-    // if the transformed is instance of ShuffleExchangeLike, so we need to remove it in AQE
-    // mode have tested gluten-it TPCH when AQE OFF
-    case ColumnarToRowExec(child: ColumnarShuffleExchangeExec) if isAdaptiveContext =>
-      replaceWithTransformerPlan(child)
-    case ColumnarToRowExec(child: ColumnarBroadcastExchangeExec) =>
-      replaceWithTransformerPlan(child)
-    case ColumnarToRowExec(child: BroadcastQueryStageExec) =>
-      replaceWithTransformerPlan(child)
-    // `InMemoryTableScanExec` internally supports ColumnarToRow
-    case ColumnarToRowExec(child: SparkPlan) if PlanUtil.isGlutenTableCache(child) =>
-      child
-    case plan: ColumnarToRowExec =>
-      transformColumnarToRowExec(plan)
-    case r: SparkPlan
-        if !r.isInstanceOf[QueryStageExec] && !r.supportsColumnar &&
-          r.children.exists(_.isInstanceOf[ColumnarToRowExec]) =>
```

Comment:
Should double-check if the check `!r.isInstanceOf[QueryStageExec]` is still needed.

Reply:
Should be safe to remove since it's impossible to have QueryStageExec's child set as C2R because we always remove the C2R in rule `RemoveTopmostColumnarToRow`. Otherwise Spark throws error from `AdaptiveSparkPlanExec#newQueryStage`.